wagtail/images/models.py

```diff
@@ -509,6 +605,89 @@ def create_rendition(self, filter: "Filter") -> "AbstractRendition":
         )
         return rendition
 
+    def create_renditions(
+        self, *filters: "Filter"
+    ) -> Dict["Filter", "AbstractRendition"]:
+        """
+        Creates multiple ``Rendition`` instances with image files reflecting the supplied
+        ``filters``, and returns them as a ``dict`` keyed by the relevant ``Filter`` instance.
+        Where suitable renditions already exist in the database, they will be returned instead,
+        so as not to create duplicates.
+
+        This method is usually called by ``Image.get_renditions()``, after first
+        checking that a suitable rendition does not already exist.
+
+        Note: If using custom image models, an instance of the custom rendition
+        model will be returned.
+        """
+        filter_map: Dict[str, Filter] = {f.spec: f for f in filters}
+
+        Rendition = self.get_rendition_model()
+        return_value: Dict[Filter, AbstractRendition] = {}
+
+        with self.open_file() as file:
+            in_memory_file = ContentFile(file.read(), name=self.file.name)
+
+        to_create = []
+        for spec, filter in filter_map.items():
+            # Use a separate Filter instance for file generation to avoid
+            # lasting references to 'in_memory_file'
+            image_file = self.generate_rendition_file(
+                Filter(spec=spec, file=in_memory_file)
+            )
+            # Reset in-memory file for next use
+            in_memory_file.seek(0)
+            # Add for bulk creation
+            to_create.append(
+                Rendition(
+                    image=self,
+                    filter_spec=filter.spec,
+                    focal_point_key=filter.get_cache_key(self),
+                    file=image_file,
+                )
+            )
+
+        # Rendition generation can take a while. So, if other processes have created
+        # identical renditions in the meantime, we should find them to avoid clashes.
+        # NB: Clashes can still occur, because there is no get_or_create() equivalent
```

Comment:
Suggestion: `select_for_update` might help here to avoid conflicts

Reply:
I'm not sure how that will help in this situation. We have a list of in-memory `Rendition` objects to save with `bulk_create`. Any items found in the following query do not need to be updated - they are used 'as is'.